src/client/datascience/jupyter/serverUriStorage.ts

```diff
@@ -124,27 +124,25 @@ export class JupyterServerUriStorage implements IJupyterServerUriStorage {
 
         if (uri === Settings.JupyterServerLocalLaunch) {
             // Just save directly into the settings
-            await this.configService.updateSetting(
-                'jupyterServerType',
-                Settings.JupyterServerLocalLaunch,
-                undefined,
-                ConfigurationTarget.Workspace
-            );
+            await this.updateServerType(Settings.JupyterServerLocalLaunch);
         } else {
             // This is a remote setting. Save in the settings as remote
-            await this.configService.updateSetting(
-                'jupyterServerType',
-                Settings.JupyterServerRemoteLaunch,
-                undefined,
-                ConfigurationTarget.Workspace
-            );
+            await this.updateServerType(Settings.JupyterServerRemoteLaunch);
 
             // Save in the storage (unique account per workspace)
             const key = this.getUriAccountKey();
             await this.encryptedStorage.store(Settings.JupyterServerRemoteLaunchService, key, uri);
         }
     }
 
+    private async updateServerType(val: string): Promise<void> {
+        if (this.workspaceService.hasWorkspaceFolders) {
+            return this.configService.updateSetting('jupyterServerType', val, undefined, ConfigurationTarget.Workspace);
```

Comment:
As `jupyterServerType` is now configured to be `application`, the target needs to be global & not workspace.

Reply:
Please see next comment.

Reply:
I guess I'm confused. I'll look it up in the VS code docs. But if the scope is global, does that mean it won't ever save it to the workspace settings?

Reply:
Yes that's what VS code docs say:

```
application - Settings that apply to all instances of VS Code and can only be configured in user settings.
resource - Resource settings, which apply to files and folders, and can be configured in all settings levels, even folder settings.
```